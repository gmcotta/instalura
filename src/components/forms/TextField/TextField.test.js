import React from 'react';
import { render, screen } from '../../../infra/test/testUtils';

import TextField from './index';

describe('<TextField />', () => {
  it('should render component', () => {
    render(
      <TextField
        placeholder="Nome"
        value="Gustavo"
        onChange={() => {}}
        name="username"
      />,
    );

    screen.debug();
    const textField = screen.getAllByPlaceholderText(/nome/i);

    expect(textField).toMatchSnapshot();
  });
});
