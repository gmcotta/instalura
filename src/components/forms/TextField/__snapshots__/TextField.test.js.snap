// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render component 1`] = `
Array [
  .c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #88989E;
  padding: 12px 16px;
  outline: 0;
  border-radius: 12px;
  margin-bottom: 0;
}

<input
    class="c0 c1"
    name="username"
    placeholder="Nome"
    type="text"
    value="Gustavo"
  />,
]
`;

exports[`<TextField /> when field is invalid should display the respective error message 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #88989E;
  padding: 12px 16px;
  outline: 0;
  border-radius: 12px;
  margin-bottom: 0;
  border-color: #dc3545;
}

.c1 + span {
  color: #dc3545;
  font-size: 11px;
}

<input
  class="c0 c1"
  name="email"
  placeholder="Email"
  type="text"
  value=""
/>
`;
